seed/src/event.rs

```diff
@@ -65,14 +81,25 @@ impl Event {
             })
             .await??;
 
+        let user = api
+            .with_storage({
```

Comment:
Why not use the same `with_storage` call above and reduce the fetches from the pool?

Reply:
It makes error handling more involved and awkward.